src/scripts/main.js

```diff
@@ -1,3 +1,37 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const tree = document.querySelector('.tree');
+
+  if (!tree) {
+    return;
+  }
+
+  // eslint-disable-next-line no-shadow
+  tree.addEventListener('click', (event) => {
+    // Перевіряємо, чи клік був по <span>
+    if (event.target.tagName === 'SPAN') {
+      const listItem = event.target.closest('li');
+
+      if (listItem) {
+        const childUl = listItem.querySelector('ul');
+
+        if (childUl) {
+          childUl.hidden = !childUl.hidden;
+        }
+      }
+    }
+  });
+
+  const headers = tree.querySelectorAll('li > span');
+
+  headers.forEach((header) => {
+    // eslint-disable-next-line no-shadow
+    const parent = header.parentElement;
+    const textNode = document.createTextNode(header.textContent);
+    const span = document.createElement('span');
+
+    span.appendChild(textNode);
+    parent.replaceChild(span, header);
+  });
```

Comment:
The logic here replaces each header `<span>` with a new `<span>` containing only text. If the original header `<span>` contains child elements (e.g., other spans or elements), they will be lost in this process. Ensure that this is the intended behavior, or modify the logic to preserve child elements if necessary.